### PR TITLE
change to lazy column, remove generics

### PR DESCRIPTION
**Pull request by Stallinator**

### CODE REVIEW

File: demo/src/main/java/de/check24/demo/MainActivity.kt [hunk 1]
```diff
@@ -62,161 +62,159 @@ class MainActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        var compareItems = mutableListOf<CompareItem>()
```
Comment (Justus-Saringer) [inline demo/src/main/java/de/check24/demo/MainActivity.kt:66]: Is it useful to switch it to val?

File: demo/src/main/java/de/check24/demo/MainActivity.kt [hunk 2]
```diff
@@ -62,161 +62,159 @@ class MainActivity : ComponentActivity() {
...
-        }
-
-        Row {
-            CompareUIItem(
+        )
+        compareItems.add(
+            CompareItem(
                 composeActivity = ComposablePasswordVisibilityToggleActivity::class.java,
                 androidUIActivity = AndroidUIPasswordVisibilityToggleActivity::class.java,
                 headline = "Password with visibility toggle"
```
Comment (Justus-Saringer) [inline demo/src/main/java/de/check24/demo/MainActivity.kt:112]: In compose the visibility toggle is in default on and in the android ui off. Just a small thing I noticed.

File: demo/src/main/java/de/check24/demo/MainActivity.kt [hunk 3]
```diff
@@ -62,161 +62,159 @@ class MainActivity : ComponentActivity() {
...
                 composeActivity = ComposableDateActivity::class.java,
                 androidUIActivity = AndroidUIDateActivity::class.java,
                 headline = "Date"
             )
-        }
+        )
 
         // endregion
 
         // region Button
```
Comment (Justus-Saringer) [inline demo/src/main/java/de/check24/demo/MainActivity.kt:132]: In the "Text" region you don't have spaces between the items, in the layout and button region you have spaces between some.

File: demo/src/main/java/de/check24/demo/MainActivity.kt (hunk 2 above)
Comment (Stallinator) [reply demo/src/main/java/de/check24/demo/MainActivity.kt:112]: actually no. the icons just behave differently. go ahead and try :) you will see that password is hidden in both from default

File: demo/src/main/java/de/check24/demo/MainActivity.kt (hunk 1 above)
Comment (Stallinator) [reply demo/src/main/java/de/check24/demo/MainActivity.kt:66]: yes

**Review by Justus-Saringer — APPROVED** [review]

**Pull request merged**